vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java

```diff
@@ -755,6 +779,98 @@ public DatePickerI18n setWeekdaysShort(List<String> weekdaysShort) {
             return this;
         }
 
+        /**
+         * Get the list of custom date formats that are used for formatting the
+         * date displayed in the text field, and for parsing the user input
+         *
+         * @return list of date patterns or null
+         */
+        public List<String> getDateFormats() {
+            return dateFormats;
+        }
+
+        /**
+         * Sets a custom date format to be used by the date picker. The format
+         * is used for formatting the date displayed in the text field, and for
+         * parsing the user input.
+         * <p>
+         * The format is a string pattern using specific symbols to specify how
+         * and where the day, month and year should be displayed. The following
+         * symbols can be used in the pattern:
+         * <ul>
+         * <li>{@code yy} - 2 digit year
+         * <li>{@code yyyy} - 4 digit year
+         * <li>{@code M} - Month
```

Comment:
I got confused for a second because you mentioned "20th of June 2021" below. This Month is a digit, right? Not some arbitrary month name?

Reply:
Good point, we can add that this will display the month as either one or two digits.

Same for the day.

Reply:
Done